ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -87,6 +87,9 @@ let create ~__context ~cluster ~host =
       | Result.Ok () ->
         Db.Cluster_host.create ~__context ~ref ~uuid ~cluster ~host ~enabled:true
           ~current_operations:[] ~allowed_operations:[] ~other_config:[];
+        (* Clustering is enabled, so fsync is enabled, sync the master DB *)
+        Helpers.call_api_functions ~__context 
```

Comment:
Could you move this before the `LocalClient.join`? I think this might be too late here as the master may have fenced when we do a `join`, and then it will have no record of this host ever being part of the cluster/pool, which is what we are trying to prevent.